TowerForge/TowerForge/Achievements/AchievementManager.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  AchievementManager.swift
+//  TowerForge
+//
+//  Created by Rubesh on 1/4/24.
+//
+
+import Foundation
+
+/// Utility class to encapsulate all the methods required for storing, loading and updating
+/// achievements and their local storage.
+class AchievementManager {
+
+    /// TODO: Standardize this to make it more easily expandable to more achievements, maybe with ObjectSet also?
+    static func incrementTotalGamesStarted() {
+        guard let achievementStorage = StorageManager
+            .getStorage(for: .achievementStorage) as? AchievementStorage else {
+
+            return
+        }
+
+        // Retrieve the specific TotalGamesAchievement
+        if let totalGamesAchievement = achievementStorage.storedObjects
+            .first(where: { $0.value is TotalGamesAchievement })?
+            .value as? TotalGamesAchievement {
+
+            totalGamesAchievement.incrementGameCount()
+            StorageManager.shared.saveToFile()
+        } else {
+            let newAchievement = TotalGamesAchievement() // Create if total games achievement doesn't already exist
+            newAchievement.incrementGameCount()
+            achievementStorage.storedObjects[newAchievement.storableName] = newAchievement
+            StorageManager.shared.saveToFile()
+        }
+    }
+
+    static func incrementTotalKillCount() {
```

Comment:
I was thinking if it s possible for this to be abstracted out of manager and the responsibility be given to the specific achievement

Reply:
I think the issue with that is each achievement would have to have a weak or unowned reference to the StorageManager or AchievementManager, which might make things complicated during encoding and decoding of achievements especially since StorageManager itself is not codable (and achievements are updated quite frequently). Even with a delegate I think the same issue of instantiating an Achievement might exist, what do you think? 

I'm thinking for now this should be sufficient, but I'm also thinking of ways to refine achievement handling (and perhaps refine the Storage Architecture as a whole) but either way I made sure that StorageManager was the only point of contact for all stored data so the underlying storage/achievements architecture can change later (if needed) without affecting the rest of the app